apm-agent-core/src/main/java/co/elastic/apm/agent/configuration/source/PropertyFileConfigurationSource.java

```diff
@@ -36,30 +39,59 @@
  */
 public final class PropertyFileConfigurationSource extends AbstractConfigurationSource {
 
+    /**
+     * Path of the configuration location
+     */
     private final String location;
+
+    /**
+     * Properties
+     */
     private Properties properties;
 
-    public PropertyFileConfigurationSource(String location) {
+    private PropertyFileConfigurationSource(String location, Properties properties) {
         this.location = location;
-        reload();
+        this.properties = properties;
     }
 
-    public static boolean isPresent(String location) {
-        return getProperties(location) != null;
+    @Nullable
+    public static ConfigurationSource fromRuntimeAttachParameters(String location) {
+        return buildSimpleSource("Attachment configuration", getPropertiesFromFilesystem(location));
     }
 
-    private static Properties getProperties(String location) {
-        if (location == null) {
+    @Nullable
+    public static ConfigurationSource fromClasspath(String location, ClassLoader classLoader) {
+        return buildSimpleSource("classpath:" + location, getPropertiesFromClasspath(location, classLoader));
+    }
+
+    @Nullable
+    public static ConfigurationSource fromFileSystem(@Nullable String location) {
+        Properties properties = getPropertiesFromFilesystem(location);
+        if (properties == null) return null;
+
+        return new PropertyFileConfigurationSource(location, properties);
+    }
+
+    private static SimpleSource buildSimpleSource(String name, Properties properties) {
+        if (properties == null) {
             return null;
         }
-        Properties props = getFromClasspath(location, ClassLoader.getSystemClassLoader());
-        if (props == null) {
-            props = getFromFileSystem(location);
+        SimpleSource source = new SimpleSource(name);
+        for (String key : properties.stringPropertyNames()) {
+            source.add(key, properties.getProperty(key));
+        }
+        return source;
+    }
+
```

Comment:
I prefer we don't have this here - those are factory methods in a type of a config source that create a different type of a config source

Reply:
Created `ConfigSources` class just for that purpose.